clients/client/src/main/java/org/projectnessie/client/auth/BasicAuthenticationProvider.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.client.auth;
+
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_PASSWORD;
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_USERNAME;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.function.Function;
+import org.projectnessie.client.http.HttpAuthentication;
+import org.projectnessie.client.http.RequestContext;
+import org.projectnessie.client.http.RequestFilter;
+
+/**
+ * HTTP BASIC authentication provider.
+ *
+ * <p>Takes parameters {@link org.projectnessie.client.NessieConfigConstants#CONF_NESSIE_USERNAME}
+ * and {@link org.projectnessie.client.NessieConfigConstants#CONF_NESSIE_PASSWORD}.
+ */
+public class BasicAuthenticationProvider implements NessieAuthenticationProvider {
+
+  public static final String AUTH_TYPE_VALUE = "BASIC";
+
+  public static NessieAuthentication of(String username, String password) {
+    return new BasicAuthentication(username, password);
+  }
+
+  @Override
+  public String getAuthTypeValue() {
+    return AUTH_TYPE_VALUE;
+  }
+
+  @Override
+  public NessieAuthentication build(Function<String, String> configSupplier) {
+    String username = configSupplier.apply(CONF_NESSIE_USERNAME);
+    String password = configSupplier.apply(CONF_NESSIE_PASSWORD);
+
+    if (username == null || password == null) {
+      throw new NullPointerException(
+          "username and password parameters must be present for auth type " + AUTH_TYPE_VALUE);
+    }
+
+    return new BasicAuthentication(username, password);
+  }
+
+  private static class BasicAuthentication implements NessieAuthentication, HttpAuthentication {
+    private final String authHeaderValue;
+
+    @SuppressWarnings("QsPrivateBeanMembersInspection")
+    private BasicAuthentication(String username, String password) {
+      String userPass = username + ':' + password;
+      byte[] encoded = Base64.getEncoder().encode(userPass.getBytes(StandardCharsets.UTF_8));
+      String encodedString = new String(encoded, StandardCharsets.UTF_8);
+      authHeaderValue = "Basic " + encodedString;
+    }
+
+    @Override
+    public RequestFilter buildRequestFilter() {
+      return new BasicHttpAuthentication(authHeaderValue);
+    }
+  }
+
+  private static class BasicHttpAuthentication implements RequestFilter {
```

Comment:
nit: Rename to `BasicHttpAuthenticationFilter` to avoid confusion with `BasicAuthentication`?